File: erisbot/src/main/java/studio/styx/erisbot/core/interfaces/CommandInterface.java
package studio.styx.erisbot.core.interfaces;

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData;

public interface CommandInterface {
    void execute(SlashCommandInteractionEvent event);

    SlashCommandData getSlashCommandData();
}

File: erisbot/src/main/kotlin/studio/styx/erisbot/core/interfaces/CommandInterface.kt
package studio.styx.erisbot.core.interfaces

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent // Import novo
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

interface CommandInterface {
    suspend fun execute(event: SlashCommandInteractionEvent)

    suspend fun onAutoComplete(event: CommandAutoCompleteInteractionEvent) {
        // Por padrão, não faz nada.
    }

    fun getSlashCommandData(): SlashCommandData
}